dll/win32/browseui/addressband.cpp

```diff
@@ -259,8 +259,63 @@ HRESULT STDMETHODCALLTYPE CAddressBand::HasFocusIO()
     return S_FALSE;
 }
 
+static WCHAR GetAccessKeyFromText(WCHAR chAccess, LPCWSTR pszText)
+{
+    for (const WCHAR *pch = pszText; *pch != UNICODE_NULL; ++pch)
+    {
+        if (*pch == L'&' && pch[1] == L'&')
+        {
+            ++pch;
+            continue;
+        }
+        if (*pch == L'&')
+        {
+            ++pch;
+            chAccess = *pch;
+            break;
```

Comment:
For @JoachimHenze : When we are in this second if-block, this is because `pch` points to '&' but the next character is not '&' (otherwise it would have been caught by the other if-block); the next character can be anything, including the terminating NULL character.
Then, `pch` is incremented (so will point to that next character or NULL), `chAccess` is set to that character (or the NULL), and the for-loop is stopped.
Again, all this is well-defined.

Reply:
Got it now. Thank you for the explanation. I missed that this part is handling the "2nd" char NULL as well.